diskstore/local_diskstore.go

```diff
@@ -211,7 +211,9 @@ func (l LocalDiskStore) OpenSnapshotVectorPartyFileForRead(table string, shard i
 	redoLogFile int64, offset uint32, batchID int, columnID int) (io.ReadCloser, error) {
 	snapshotFilePath := GetPathForTableSnapshotColumnFilePath(l.rootPath, table, shard, redoLogFile, offset, batchID, columnID)
 	f, err := os.OpenFile(snapshotFilePath, os.O_RDONLY, 0644)
-	if err != nil {
+	if os.IsNotExist(err) {
```

Comment:
why not just directly return os.ErrNotExist

Reply:
this is to follow the other convention, we not treat this is error

Reply:
this might be a different case, for user still check for reader == nil to get the os.ErrNonExist.  But for most metastore cases, user does not care about the os.ErrNonExist error

Reply:
then we need to change the behavior of both OpenVectorPartyFileForRead and OpenSnapshotFileForRead

Reply:
doesn't seem to change the behavior, before the change, all errors including nonexist error return to the user.
after the change, we just don't wrap the NonExistErr with stack error if user need to know this specific error